### ISSUE TITLE
DNS TXT record length too long for p521 curve

### ISSUE THREAD

**Issue by jekriske-lilly** [upspin/upspin#400]
Many DNS implementations have a 255 character limit for a TXT record. 

If a user account uses a p521 curve and one attempts to `upspin setupdomain`, even with the default curve, it results in a TXT entry that's too large.

In my opinion the best solution is Curve 25519 support as listed https://github.com/upspin/upspin/issues/206 and if the support for large NIST keys remains, perhaps this can span multiple TXT records or some other possible solution??

**Comment by shoeper** [comment]
https://tools.ietf.org/html/rfc4408#section-3.1.3

**Comment by jekriske-lilly** [comment]
Thanks, exactly. As described in the document multiple strings would probably work.

**Comment by adg** [comment]
Maybe we should make the TXT record be a SHA256 sum of the signature, so that it's always the same length regardless of the curve used. @n2vi would that still be secure?

**Comment by n2vi** [comment]
I'm away celebrating my 35th anniversary with Brenda and have no wifi, so
this phone-based email will have to be short until I get back to town.



Unless somebody knows something I don't, the reason to use p521 is to get
the equivalent of 256-bit symmetric key length.  So I'm puzzled about the
suggestion to use 25519 as an alternative.  The crypto lengths don't seem
to match.   We do already use 128 entropy and 256 AES, so agree that there
are nuances in picking lengths.  Just saying this needs more discussion
than I can do from here.

How urgent is this?  Don't we allow signing up with p256, which is
generally regarded as strong enough for the moment, then rotating key to
p521 after setup is complete?   I'm not in a position to test.

Hash lengths would normally be twice the symmetric key length following the
birthday collision rule, hence SHA-512.   Anyway, if you hash the signature
how can it be verified?

So in short I'd use multiple TXT strings as proposed, if length is an
urgent issue.

**Comment by adg** [comment]
@n2vi but what do you think about using a hash of the signature instead of the signature itself? That would mean we'd only need one TXT record and reduce the likelihood of mistakes.

**Comment by n2vi** [comment]
"if you hash the signature how can it be verified?"    might be clarified as:
If you give me a file and a signature of the file and the public key used for signing, I know how to verify.
If you replace the signature with hash(signature) then I don't know how to check it.

Or maybe I don't understand what you mean by hashing the signature?
